watson-developer-cloud/discovery-components
# feat: Show more modal

#### What do these changes do/fix?
These changes add in a modal that opens when the `Show more` is clicked for a facet that has greater than 10 values. In the modal, the user can select and deselect facets and then save those selections.

Note that the styles for the left padding for the Carbon buttons in the modal will need to be edited in Tooling when these changes are updated there to not remove the padding for buttons in this modal.

#### How do you test/verify these changes?
This can be tested by linking to Tooling and playing around with it there, running the example app and taking a look at it there, and it can also be seen in Storybook. Also take a look at tests and confirm they're testing everything we want to.

Some things to take a look at include that it's working as expected for single-select and multi-select, that the modal only saves selections on `Save`, not on `Cancel` or closing the modal, that the modal only shows when there are greater than 10 facet values, that it's working for categories within entities facets.

#### Have you documented your changes (if necessary)?
There's an example in Storybook of a facet (`products`) with more than 10 values that opens to the `Show more` modal.

There's also an example of a category (`Quantity`) within an entities facet with more than 10 values that opens to show the `Show more` modal.

#### Are there any breaking changes included in this pull request?
no
